### ISSUE TITLE
Throw error when formula has parameters that are not available

### ISSUE THREAD

**Issue by hguturu** [matthewwardrop/formulaic#159]
E..g.
```py
formulaic.model_matrix(["1+BinGrpe"],phenotypes)
where
phenotypes =  all_phenotypes = pd.DataFrame({ "AltGrp": [1, 0, 0, 1, 0, 1], "BinGrp": [0, 0, 0, 1, 1, 1], "ContGrp" : [1,2,3,4,5,6]})
```
yields
```
   Intercept
0        1.0
1        1.0
2        1.0
3        1.0
4        1.0
5        1.0
```
Where the typo in `BinGrpe` is silently ignored. I think it would be safer if an error or warning is thrown here.

**Comment by matthewwardrop** [comment]
This is a regression introduced in a recent release. Will make sure this get's patched.

**Comment by matthewwardrop** [comment]
Fixed by 2f0a480641aba40d0fc3783797110165baeaddf0
